micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmVirtualThreadEventMetrics.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jvm;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import jdk.jfr.consumer.RecordingStream;
+
+import static java.util.Collections.emptyList;
+
+public class JvmVirtualThreadEventMetrics implements MeterBinder, AutoCloseable {
+
+    private final Iterable<Tag> tags;
+
+    private final RecordingStream recordingStream;
+
+    public JvmVirtualThreadEventMetrics() {
+        this(emptyList());
+    }
+
+    public JvmVirtualThreadEventMetrics(Iterable<Tag> tags) {
+        this.tags = tags;
+        this.recordingStream = new RecordingStream();
+
+        recordingStream.enable("jdk.VirtualThreadPinnedEvent");
+        recordingStream.enable("jdk.VirtualThreadSubmitFailedEvent");
```

Comment:
I'm not familiar with the right `RecordingStream` usage, but I was looking at [this blog](https://mikemybytes.com/2024/04/17/continuous-monitoring-of-pinned-threads-with-spring-boot-and-jfr/), and it made me wonder: do we not need to start the stream? And should we not set a max age so we don't accumulate events indefinitely?

Reply:
hey @shakuzen,

Do you think we need to expose maxAge duration as a constructor parameter?

Reply:
Good question. I had the same thought, but I'm not sure due to my lack of familiarity with JFR and in particular `RecordingStream`. I don't understand what this max age means in terms of the event listeners we register. I assume the event listeners are run nearly instantly after an event is recorded. So I'm not sure the max age could affect them. Or will max age effectively filter out events with a duration longer than the max age duration?